garak/generators/bedrock.py

```diff
@@ -0,0 +1,258 @@
+        if hasattr(self, "temperature") and self.temperature is not None:
+            inference_config["temperature"] = float(self.temperature)
+        if hasattr(self, "max_tokens") and self.max_tokens is not None:
+            inference_config["maxTokens"] = int(self.max_tokens)
+        if hasattr(self, "top_p") and self.top_p is not None:
+            inference_config["topP"] = float(self.top_p)
+        if hasattr(self, "stop") and self.stop and len(self.stop) > 0:
+            inference_config["stopSequences"] = list(self.stop)
+
+        # Prepare API call arguments
+        call_args = {
+            "modelId": self.name,
+            "messages": messages,
+        }
+
+        # Only add inferenceConfig if it has parameters
+        if inference_config:
+            call_args["inferenceConfig"] = inference_config
+
+        try:
+            # Call the Converse API
+            response = self.client.converse(**call_args)
+
+            # Extract text from response
+            if not response or "output" not in response:
+                logging.error("Malformed response from Bedrock: missing 'output' field")
+                return [None]
+
+            if "message" not in response["output"]:
+                logging.error(
+                    "Malformed response from Bedrock: missing 'message' in output"
+                )
+                return [None]
+
+            message = response["output"]["message"]
+            if "content" not in message or not message["content"]:
+                logging.error(
+                    "Malformed response from Bedrock: missing or empty 'content' in message"
+                )
+                return [None]
+
+            # Extract text from the first content block
+            content_block = message["content"][0]
+            if "text" not in content_block:
+                logging.error(
+                    "Malformed response from Bedrock: missing 'text' in content block"
+                )
+                return [None]
+
+            text = content_block["text"]
+            return [Message(text=text)]
+
+        except Exception as e:
+            try:
+                from botocore.exceptions import ClientError
+
+                if isinstance(e, ClientError):
+                    error_code = e.response.get("Error", {}).get("Code", "")
+                    error_message = e.response.get("Error", {}).get("Message", "")
+
+                    logging.error(f"Bedrock API error [{error_code}]: {error_message}")
+
+                    # Check if we should retry (throttling and service unavailable are transient)
+                    if error_code in [
+                        "ThrottlingException",
+                        "ServiceUnavailableException",
+                    ]:
+                        raise garak.exception.GarakBackoffTrigger from e
+
+                    # For non-retriable errors (validation, access denied), return None
+                    return [None]
```

Comment:
Based on the comment here, will the `giveup` call to `_should_not_retry()` ever return `True`?